java/ql/src/experimental/Security/CWE/CWE-347/Example.java

```diff
@@ -0,0 +1,59 @@
+package com.example.JwtTest;
+
+import java.io.*;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+import javax.crypto.KeyGenerator;
+import javax.servlet.http.*;
+import javax.servlet.annotation.*;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+@WebServlet(name = "Jwt", value = "/Auth")
+public class auth0 extends HttpServlet {
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {}
+
+  final String JWT_KEY = "KEY";
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // OK
+    String JwtToken1 = request.getParameter("JWT1");
+    decodeToken(JwtToken1);
+    try {
+      verifyToken(JwtToken1, getSecureRandomKey());
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
+    }
+
+    // only decode, no verification
+    String JwtToken2 = request.getParameter("JWT2");
+    decodeToken(JwtToken2);
```

Comment:
Since nobody is using the result of `decodeToken` why does it matter whether we verify it or not?

Reply:
It is a good point made by @intrigus-lgtm. As you see in the similar query for JS it can be decisive factor if the decoded payload is used or not. I think you should update your tests and query to raise the alert if it is used and do not raise if it is discarded.

Reply:
BTW, I think your query already discards the alert if the result of decode is not used, but it would be good to confirm it with tests.

Reply:
I don't think you need changes to the existing codeql query. Have you tried just adding two tests: one that doesn't do anything with the result of decodeToken (should not alert) and another one that calls `getClaim` for example?